### PR TITLE
Moving function and function_ref into namespace hpx

### PR DESCRIPTION
**Pull request by hkaiser**
- rename unique_function to hpx::move_only_function

### CODE REVIEW

**Review by NK-Nikunj — COMMENTED** [review]
Quick question: What does `Serializability` mean here? Is it to distinguish a function pointer with an action type?

File: libs/core/functional/tests/unit/contains_test.cpp [hunk 1]
```diff
@@ -74,7 +74,7 @@ static void target_test()
 
 //static void equal_test()
 //{
-//    hpx::util::function_nonser<int()> f;
+//    hpx::function<int()> f;
 //
 //    f = &forty_two;
```
[the remark this answers is not in the document]
Comment (hkaiser) [reply libs/core/functional/tests/unit/contains_test.cpp:79]: Good question, will investigate ;-)

**Comment by hkaiser** [comment]
> Quick question: What does `Serializability` mean here? Is it to distinguish a function pointer with an action type?

It means whether an object of type `hpx::function` will be serializable or not.

File: libs/core/functional/tests/unit/contains_test.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (hkaiser) [reply libs/core/functional/tests/unit/contains_test.cpp:79]: @NK-Nikunj those tests are leftovers and need to be removed. Originally these tests were taken from Boost. `boost::function` supports more things than what `std::function` (and therefore `hpx::function`) does. I will remove the commented sections of code. Thanks for noticing!

[the remark this answers is not in the document]
Comment (hkaiser) [reply libs/core/functional/tests/unit/contains_test.cpp:79]: See https://github.com/STEllAR-GROUP/hpx/pull/5787

**Pull request merged**